administrative-sdk/pronunciationChallenge.js

```diff
@@ -115,150 +77,86 @@ class PronunciationChallenge {
-   *
-   * @callback Sdk~getPronunciationChallengeCallback
-   * @param {its.PronunciationChallenge} challenge Retrieved pronunciation challenge domain model instance.
-   */
-  getPronunciationChallengeCallback(challenge) {}
-
-  /**
-   * Error callback used by getPronunciationChallenge.
-   *
-   * @callback Sdk~getPronunciationChallengeErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  getPronunciationChallengeErrorCallback(errors) {}
-
-  /**
    * Get a pronunciation challenge.
    *
    * @param {string} organisationId Specify an organisation identifier.
    * @param {string} challengeId Specify a pronunciation challenge identifier.
    * @param {Sdk~getPronunciationChallengeCallback} [cb] The callback that handles the response.
    * @param {Sdk~getPronunciationChallengeErrorCallback} [ecb] The callback that handles the error response.
    */
-  static getPronunciationChallenge(
-    connection, organisationId, challengeId, cb, ecb) {
-    var _cb = function(data) {
-      var challenge = new PronunciationChallenge(organisationId, data.id,
-        data.transcription);
-      challenge.created = new Date(data.created);
-      challenge.updated = new Date(data.updated);
-      challenge.referenceAudioUrl = data.referenceAudioUrl;
-      challenge.status = data.status;
-      if (cb) {
-        cb(challenge);
-      }
-    };
-
+  static getPronunciationChallenge(connection, organisationId, challengeId) {
     var url = connection.settings.apiUrl + '/organisations/' +
       organisationId + '/challenges/pronunciation/' + challengeId;
-    connection._secureAjaxGet(url, _cb, ecb);
+    return connection._secureAjaxGet(url)
+      .then(data => {
+        var challenge = new PronunciationChallenge(organisationId, data.id,
+          data.transcription);
+        challenge.created = new Date(data.created);
+        challenge.updated = new Date(data.updated);
+        challenge.referenceAudioUrl = data.referenceAudioUrl;
+        challenge.status = data.status;
+        return challenge;
+      });
   }
 
   /**
-   * Callback used by listPronunciationChallenges.
-   *
-   * @callback Sdk~listPronunciationChallengesCallback
-   * @param {its.PronunciationChallenge[]} challenges Retrieved pronunciation challenge domain model instances.
-   */
-  listPronunciationChallengesCallback(challenges) {}
-
-  /**
-   * Error callback used by listPronunciationChallenges.
-   *
-   * @callback Sdk~listPronunciationChallengesErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  listPronunciationChallengesErrorCallback(errors) {}
-
-  /**
    * List all pronunciation challenges in the organisation.
    *
    * @param {string} organisationId Specify an organisation identifier.
    * @param {Sdk~listPronunciationChallengesCallback} cb The callback that handles the response.
    * @param {Sdk~listPronunciationChallengesErrorCallback} [ecb] The callback that handles the error response.
```

Comment:
Missing documentation update. Callbacks are no longer taken and a promise is returned.